### PR TITLE
Initial draft content for parallel execution

### PR DESCRIPTION
**Pull request by seanmooretechwriter**
Initial draft content for review.

### CODE REVIEW

File: docs/developer-guide/parallel-execution.md
Comment (maoueh) [inline docs/developer-guide/parallel-execution.md:2]: Shouldn't be 
```suggestion
description: StreamingFast Substreams Parallel Execution
```

File: docs/developer-guide/parallel-execution.md
Comment (maoueh) [inline docs/developer-guide/parallel-execution.md:7]: `pre-compute` is not accurate enough because when doing backward parallel execution, we are not pre-computing.

File: docs/developer-guide/parallel-execution.md
Comment (maoueh) [inline docs/developer-guide/parallel-execution.md:9]: I'm quite unsure if those "marketing" like sentence have their place in the documentation. At least should be re-worded maybe we just say that full chain processing can be achieved in ~15 minutes (I was able in the past to perform a full chain processing in 15 minutes).

Or we eliminate it completely.

Comment (maoueh) [reply docs/developer-guide/parallel-execution.md:9]: I'm talking specifically about `StreamingFast was able to use 120 concurrent workers to completely process and store all of the ERC20/ERC721/ERC1155 token transfers on the Ethereum Mainnet in nearly 45 minutes!`. So with the first sentence, I think we should still say something about ability to fully process chain in minutes or something like that.

File: docs/developer-guide/parallel-execution.md
Comment (maoueh) [inline docs/developer-guide/parallel-execution.md:17]: Let's be agnostic of a specific chain and let's just say `Assuming a chain that has 16,000,000 blocks`, it translates to ...` something like this.

Comment (maoueh) [reply docs/developer-guide/parallel-execution.md:17]:
```
... it depends on the capabilities of the service provider StreamingFast ...
``` 

Seems it's missing a `.` here?

File: docs/developer-guide/parallel-execution.md
Comment (maoueh) [inline docs/developer-guide/parallel-execution.md:19]: I've change my mind a bit, let's use `Backward parallel execution` to discuss about this. Let's also add a paragraph to quickly explain `Forward parallel execution`.

File: docs/developer-guide/parallel-execution.md
Comment (maoueh) [inline docs/developer-guide/parallel-execution.md:21]: Feels this should be much higher in the document, feels weird at the end like that.

**Review by maoueh — CHANGES_REQUESTED** [review]
Can we also bring the graphic in https://github.com/streamingfast/substreams/blob/develop/docs/release-notes/change-log.md#parallel-processing and bring in explanations of what we see in the graphic.

File: docs/developer-guide/parallel-execution.md
Comment (seanmooretechwriter) [reply docs/developer-guide/parallel-execution.md:2]: Changed to: StreamingFast Substreams parallel execution, per the style guide. The update will be in the next commit.

File: docs/developer-guide/parallel-execution.md
Comment (seanmooretechwriter) [reply docs/developer-guide/parallel-execution.md:7]: I've removed the initial sentence and left the rest of the content. The update will be in the next commit.

File: docs/developer-guide/parallel-execution.md
Comment (seanmooretechwriter) [reply docs/developer-guide/parallel-execution.md:9]: Yes, this is correct. Per the style guide, we should not include any marketing in the documentation. In addition, we are **NOT** to use words like easy, fast, quick, etc.

This sentence has been removed. The update will be in the next commit.

File: docs/developer-guide/parallel-execution.md
Comment (seanmooretechwriter) [reply docs/developer-guide/parallel-execution.md:17]: Updates made and will be available in the next commit.

File: docs/developer-guide/parallel-execution.md
Comment (seanmooretechwriter) [reply docs/developer-guide/parallel-execution.md:21]: Content moved directly below the original first paragraph. Changes will be available in the next commit.

File: docs/developer-guide/parallel-execution.md
Comment (seanmooretechwriter) [reply docs/developer-guide/parallel-execution.md:19]: Please review the latest commit addressing all other change requests for this PR. Part of the content referenced here has now been moved, per request.

Please let me know if these are subtitle/headers you would like to add to the existing content or new sections. If they are new sections please help locate or provide content for them.

**Comment by seanmooretechwriter** [comment]
All changes except the one referencing the reintroduction of "_backward/forward_" processing language have been addressed, committed, and pushed.

File: docs/developer-guide/parallel-execution.md
Comment (maoueh) [inline docs/developer-guide/parallel-execution.md:7]:
```suggestion
Parallel execution is the process of a Substreams module's code executing multiple segments of blockchain data simultaneously. Substreams modules can be executed in parallel, rapidly producing data for consumption in end-user applications. Parallel Execution enables Substreams' highly efficient blockchain data processing capabilities.
```

File: docs/developer-guide/parallel-execution.md
Comment (maoueh) [inline docs/developer-guide/parallel-execution.md:9]: It can happen in two ways, the one described here but also in forward parallel execution, the first sentence gives the impression that only the former is true.

I see the last sentence of this paragraph talking about full range, but it's really not clear. I think a re-work is neeed.

File: docs/developer-guide/parallel-execution.md
Comment (maoueh) [reply docs/developer-guide/parallel-execution.md]: I've commented back in the my follow-up review.

**Review by maoueh — CHANGES_REQUESTED** [review]

**Comment by maoueh** [comment]
@seanmooretechwriter Please also take a look the general comment I made:

> Can we also bring the graphic in https://github.com/streamingfast/substreams/blob/develop/docs/release-notes/change-log.md#parallel-processing and bring in explanations of what we see in the graphic.

File: docs/developer-guide/parallel-execution.md
Comment (seanmooretechwriter) [reply docs/developer-guide/parallel-execution.md:7]: These updates have been made and will be available for review in the next commit/push.

File: docs/developer-guide/parallel-execution.md
Comment (seanmooretechwriter) [reply docs/developer-guide/parallel-execution.md:9]: First sentence updated. I removed the last sentence. I do not know how to improve it.

"Parallel execution is performed for the full range of blocks when Substreams is in production mode."

Please let me know if you have any input to reword this for clarity and technical accuracy.

Thank you!

The changes will be available for review in the next commit/push.

**Comment by seanmooretechwriter** [comment]
These changes have been addressed. I've added the diagram graphic as well. Please review for technical accuracy and let me know what we can do to build out the graphic explanation/description, I provided what I understand to this point and am able to deduce from the image.

Thank you!

Change have been commited/pushed.

File: docs/developer-guide/parallel-execution.md
Comment (maoueh) [inline docs/developer-guide/parallel-execution.md:23]: `Parallel processing` -> `Parallel execution`, `forward processing` -> `foward parallel execution`

File: docs/developer-guide/parallel-execution.md
Comment (maoueh) [inline docs/developer-guide/parallel-execution.md:21]: Don't `diagram` is needed at the end.

**Review by maoueh — CHANGES_REQUESTED** [review]
Final comments, we are good after that, do not merge, versions of Substreams is still unreleased for this.

File: docs/developer-guide/parallel-execution.md
Comment (seanmooretechwriter) [reply docs/developer-guide/parallel-execution.md:23]: This has been updated and will be available for review in the next commit/push.

File: docs/developer-guide/parallel-execution.md
Comment (seanmooretechwriter) [reply docs/developer-guide/parallel-execution.md:21]: This has been updated and will be available for review in the next commit/push.

**Comment by seanmooretechwriter** [comment]
The latest changes have been addressed. The changes have been committed/pushed. Nothing has been merged.

**Pull request merged**
